src/phantasm-renderer/backend/gpu_info.cc

```diff
@@ -0,0 +1,116 @@
+#include "gpu_info.hh"
+
+#include <iostream>
+
+namespace
+{
+constexpr char const* get_preference_literal(pr::backend::adapter_preference pref)
+{
+    switch (pref)
+    {
+    case pr::backend::adapter_preference::first:
+        return "first";
+    case pr::backend::adapter_preference::integrated:
+        return "integrated";
+    case pr::backend::adapter_preference::highest_vram:
+        return "highest vram";
+    case pr::backend::adapter_preference::explicit_index:
+        return "explicit index";
+    case pr::backend::adapter_preference::highest_feature_level:
+        return "highest feature level";
+    }
+}

```

Comment:
maybe we can think about some semi-magic reflector introspect solution for enum <-> string